tests/Pregel/ConductorStateTest.cpp

```diff
@@ -0,0 +1,210 @@
+
+struct LookupInfoMock : conductor::LookupInfo {
+  LookupInfoMock(std::vector<std::string> serverIDs) : _servers(serverIDs) {}
+  ~LookupInfoMock() = default;
+
+  [[nodiscard]] auto getServerMapVertices() const -> ServerMapping override {
+    ServerMapping mapping{};
+    for (auto const& serverID : _servers) {
+      mapping.emplace(serverID,
+                      std::map<std::string, std::vector<std::string>>{});
+    }
+    return mapping;
+  }
+
+  // edges related methods:
+  [[nodiscard]] auto getServerMapEdges() const -> ServerMapping override {
+    ServerMapping mapping{};
+    for (auto const& serverID : _servers) {
+      mapping.emplace(serverID,
+                      std::map<std::string, std::vector<std::string>>{});
+    }
+    return mapping;
+  }
+
+  // both combined (vertices, edges) related methods:
+  [[nodiscard]] auto getAllShards() const -> ShardsMapping override {
+    return {};
+  }
+
+  [[nodiscard]] auto getCollectionPlanIdMapAll() const
+      -> CollectionPlanIDMapping override {
+    return {};
+  }
+
+  std::vector<std::string> _servers;
+};
+
+TEST(ConductorStateTest,
+     must_always_be_initialized_with_initial_execution_state) {
+  std::vector<std::string> emptyServers{};
+  auto cState = ConductorState(ExecutionSpecifications(),
+                               std::make_unique<LookupInfoMock>(emptyServers));
+  ASSERT_EQ(cState._executionState->name(), "initial");
+}
+
+TEST(CreateWorkersStateTest, creates_as_many_messages_as_required_servers) {
+  std::vector<std::vector<std::string>> amountOfServers = {
+      {}, {"ServerA"}, {"ServerA", "ServerB"}};
+
+  for (auto const& servers : amountOfServers) {
+    auto cState = ConductorState(ExecutionSpecifications(),
+                                 std::make_unique<LookupInfoMock>(servers));
+    auto createWorkersState = CreateWorkers(cState);
+    auto msgs = createWorkersState.messages();
+    ASSERT_EQ(msgs.size(), servers.size());
+
+    for (auto const& serverName : servers) {
+      ASSERT_TRUE(msgs.contains(serverName));
+    }
+  }
+}
+
+TEST(CreateWorkersStateTest, check_that_conductor_creates_all_worker_pids) {
+  std::vector<std::string> servers = {"ServerA", "ServerB", "ServerC"};
+
+  auto cState = ConductorState(ExecutionSpecifications(),
+                               std::make_unique<LookupInfoMock>(servers));
+  auto createWorkers = CreateWorkers(cState);
+  auto msgs = createWorkers.messages();
+
+  auto created = arangodb::ResultT<message::WorkerCreated>();
+  ASSERT_TRUE(created.ok());
+  {
+    createWorkers.receive(
+        {.server = servers.at(0), .database = databaseName, .id = {0}},
+        created);
+    createWorkers.receive(
+        {.server = servers.at(1), .database = databaseName, .id = {1}},
+        created);
+    createWorkers.receive(
+        {.server = servers.at(2), .database = databaseName, .id = {2}},
+        created);
+  }
+
+  ASSERT_EQ(createWorkers.conductor._workers.size(), servers.size());
```

Comment:
this is the same check as below because createWorkers.conductor is reference to cState, so I would delete it

Reply:
Thank you, applied suggestion.